chips/apollo3/src/clkgen.rs

```diff
@@ -0,0 +1,63 @@
+//! Power Reset Clock Interrupt controller driver.
+
+use kernel::common::registers::{register_structs, ReadWrite};
+use kernel::common::StaticRef;
+
+pub static mut CLKGEN: ClkGen = ClkGen::new(CLKGEN_BASE);
+
+const CLKGEN_BASE: StaticRef<ClkGenRegisters> =
+    unsafe { StaticRef::new(0x4000_4000 as *const ClkGenRegisters) };
+
+register_structs! {
+    pub ClkGenRegisters {
+        (0x00 => calxt: ReadWrite<u32>),
+        (0x04 => calrc: ReadWrite<u32>),
+        (0x08 => acalctr: ReadWrite<u32>),
+        (0x0c => octrl: ReadWrite<u32>),
+        (0x10 => clkout: ReadWrite<u32>),
+        (0x14 => clkkey: ReadWrite<u32>),
+        (0x18 => cctrl: ReadWrite<u32>),
+        (0x1c => status: ReadWrite<u32>),
+        (0x20 => hfadj: ReadWrite<u32>),
+        (0x24 => _reserved0),
+        (0x28 => clockenstat: ReadWrite<u32>),
+        (0x2c => clocken2stat: ReadWrite<u32>),
+        (0x30 => clocken3stat: ReadWrite<u32>),
+        (0x34 => freqctrl: ReadWrite<u32>),
+        (0x38 => _reserved1),
+        (0x3c => blebucktonadj: ReadWrite<u32>),
+        (0x40 => _reserved2),
+        (0x100 => intrpten: ReadWrite<u32>),
+        (0x104 => intrptstat: ReadWrite<u32>),
+        (0x108 => intrptclr: ReadWrite<u32>),
+        (0x10c => intrptset: ReadWrite<u32>),
+        (0x110 => @END),
+    }
+}
+
+pub enum ClockFrequency {
+    FreqMax,
```

Comment:
Is this 48MHz or 96 MHz? Not clear to me because of how apollo refers to 96 MHz as the TurboSpot frequency which seems to apply it can't run at that speed full time

Reply:
I'm pretty sure it's 48MHz, I'll update this.